### PR TITLE
upgrade versions: - nn 97.0.0; - maps 10.5.0-rc.1;

### PR DESCRIPTION
**Pull request by RingerJK**

### CODE REVIEW

File: libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSourceTest.kt [hunk 1]
```diff
@@ -125,7 +125,7 @@ class MapboxNavigationViewportDataSourceTest {
 
     @Before
     fun setup() {
-        mockkStatic(DirectionsRoute::toNavigationRoute)
+        mockkStatic("com.mapbox.navigation.base.route.NavigationRouteEx")
```
Comment (LukasPaczos) [inline libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSourceTest.kt:128]: Does the function reference not work?

**Comment by LukasPaczos** [comment]
Where's the Kotlin stdlib update coming from? We might need to avoid it for now until we setup a strategy across the SDKs.

File: libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSourceTest.kt (hunk 1 above)
Comment (RingerJK) [reply libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSourceTest.kt:128]: it's not working anymore, when a function is overloaded the compiler cannot resolve which methods must be used

**Comment by RingerJK** [comment]
> Where's the Kotlin stdlib update coming from? We might need to avoid it for now until we setup a strategy across the SDKs.

kotlin is not updated in this PR. Initially, I added an update of the kotlin version, but later separate into another PR https://github.com/mapbox/mapbox-navigation-android/pull/5743. Changed name of the PR

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt [hunk 2]
```diff
@@ -45,16 +47,20 @@ fun NavigationRoute.updateLegAnnotations(
     val refreshedResponse = directionsResponse.toBuilder()
         .routes(refreshedRoutes)
         .build()
+    HttpServiceFactoryWrapper
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt:50]: can be removed?

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt [hunk 3]
```diff
@@ -244,7 +307,32 @@ fun List<NavigationRoute>.toDirectionsRoutes() = map { it.directionsRoute }
  *
  * **Avoid using this mapper and instead try using APIs that accept [NavigationRoute] type where possible.**
  */
-fun List<DirectionsRoute>.toNavigationRoutes() = map { it.toNavigationRoute() }
+@Deprecated(
+    "NavigationRoute requires routerOrigin. If RouterOrigin is not set uses RouterOrigin.Custom()",
+    ReplaceWith("toNavigationRoutes(routerOrigin)")
+)
+fun List<DirectionsRoute>.toNavigationRoutes() =
+    map { it.toNavigationRoute(RouterOrigin.Custom()) }
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:315]: Wondering if we can be smarter about it. This is a compatibility function so I think we should make the best effort to figure out the source ourselves and only fallback to `Custom` when needed.

For example, if `route.uuid` is available, it's an `offboard` route.

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt [hunk 4]
```diff
@@ -260,11 +348,37 @@ fun List<DirectionsRoute>.toNavigationRoutes() = map { it.toNavigationRoute() }
...
  * **Avoid using this mapper and instead try using APIs that accept [NavigationRoute] type where possible.**
  */
+@Deprecated(
+    "NavigationRoute requires routerOrigin. If RouterOrigin is not set uses RouterOrigin.Custom()",
+    ReplaceWith("toNavigationRoute(routerOrigin)")
+)
 fun DirectionsRoute.toNavigationRoute(): NavigationRoute = this.toNavigationRoute(
-    NativeRouteParserWrapper
+    NativeRouteParserWrapper,
+    RouterOrigin.Custom(),
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:357]: Same as above.

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt (hunk 3 above)
Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:315]: I've already sync with NN, and there's no good way to check what's route origin, it also might be Custom and we make change it with a false prediction

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt (hunk 2 above)
Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt:50]: unnecessary changes, removed

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt (hunk 3 above)
Comment (LukasPaczos) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:315]: That's where we have the other method that does accept origin as parameter.

This parameter-less function is here to provide our best compatibility with the old router interfaces (and we can document it this way), so I still think we should at least check the UUID.

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:315]: that's actually a good point of investigation: what if a user stored the route locally (as we do in mobile-metrics actually) and set it to the SDK with `uuid` and `RouterOrigin.OffBoard` NN will assume one as `Offboard` router with annotations and incidents and trying to update it but cannot.

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:315]: > This parameter-less function is here to provide our best compatibility with the old router interfaces (and we can document it this way), so I still think we should at least check the UUID.

We might assume the following cases:
- a route has UUID and it contains `@local` string -> route is onBoard
- a route has UUID -> offBoard
- else -> Custom

cc @LukasPaczos @mskurydin to check if it makes sense

Comment (mskurydin) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:315]: The UUID is not well documented and its contents is subject to change. Not sure we can rely on it for this.

Comment (LukasPaczos) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:315]: As discussed internally, let's merge as is. The only thing we'll need to remember is not make the origin a strong condition for other features :+1:

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,12 @@ Mapbox welcomes participation and contributions from everyone.
 #### Features
 
 #### Bug fixes and improvements
+- :warning: Deprecated `NavigationRoute` creation and mapping functions in favor of equivalents that take `RouterOrigin` as a parameter. If not provided, `RouterOrigin.Custom()` is used. [#5738](https://github.com/mapbox/mapbox-navigation-android/pull/5738)
+  - method `NavigationRoute#create(DirectionsResponse, RouteOptions)` by `NavigationRoute#create(DirectionsResponse, RouteOptions, RouterOrigin)`;
+  - method `NavigationRoute#create(String, String)` by `NavigationRoute#create(String, String, RouterOrigin)`;
+  - extension `List<DirectionsRoute>#toNavigationRoutes()` by `List<DirectionsRoute>#toNavigationRoutes(RouterOrigin)`;
+  - extension `DirectionsRoute#toNavigationRoute()` by `DirectionsRoute#toNavigationRoute(RouterOrigin)`.
+- Fixed runtime exception on [replacing the default logger module](https://docs.mapbox.com/android/navigation/guides/get-started/modularization/#logger) during library loading. [#5738](https://github.com/mapbox/mapbox-navigation-android/pull/5738)
```
Comment (LukasPaczos) [inline CHANGELOG.md:15]:
```suggestion
- Fixed an issue where [replacing the default logger module](https://docs.mapbox.com/android/navigation/guides/get-started/modularization/#logger) was throwing a runtime exception during library loading. [#5738](https://github.com/mapbox/mapbox-navigation-android/pull/5738)
```

**Review by LukasPaczos — APPROVED** [review]

**Pull request merged**